### ISSUE TITLE
Add a logarithmic loss type to `LossType` (matches matplotlib-venn / set-diagrams objectives)

### ISSUE THREAD

**Issue by jolars** [jolars/eunoia#96]
## Summary

Add a **logarithmic absolute-difference loss** to `LossType` (e.g. `LogSumAbsolute`, plus a smooth surrogate `SmoothLogSumAbsolute`):

```
loss = Σ |log(1 + f_i) − log(1 + t_i)|  /  Σ |log(1 + t_i)|
```

where `f_i` is the fitted area and `t_i` the requested value of region `i` (normalized like `SumAbsolute`).

## Motivation

While building a fair "matched-objective" benchmark of eunoia against the other Python area-proportional fitters (in `eunoia-py`), it turned out that the only way to compare optimizers fairly is to group packages by the loss they actually minimize and compare within each group on that loss. Two of the competitors minimize a logarithmic L1 cost that eunoia currently has no equivalent for:

- **`matplotlib-venn`** (`venn3`, default layout): minimizes `Σ |log(1 + target) − log(1 + actual)|` over the 7 regions (its `WeightedAggregateCost(transform_fn = lambda x: np.log(1 + x))`).
- **`matplotlib-set-diagrams`** (`cost_function_objective="logarithmic"`): minimizes `Σ |log(1 + actual) − log(1 + target)|`.

eunoia's `LossType` today covers `SumSquared`, `SumAbsolute`, `{Sum,Max}{Absolute,Squared}`, `{Sum}{Absolute,Squared}RegionError`, `RootMeanSquared`, `Stress`, and `DiagError` (`crates/eunoia/src/loss.rs`) — but nothing logarithmic. So eunoia cannot enter the head-to-head group that `venn3`'s *default* objective defines, and that comparison currently has to exclude eunoia.

The log transform also has a natural property worth supporting on its own merits: it compresses dynamic range, so large regions don't dominate the fit the way they do under raw `SumSquared`/`SumAbsolute` — useful for specs with orders-of-magnitude size variation.

## Proposed implementation

- Add `LogSumAbsolute` to the `LossType` enum and a matching arm in the loss `compute` match (~`crates/eunoia/src/loss.rs:360+`), mirroring `SumAbsolute` but with `log1p` applied to both `f` and `t`.
- `abs` is non-smooth, so add a `SmoothLogSumAbsolute { eps }` surrogate reusing the existing `smooth_abs`, analogous to the `SumAbsolute` / `SmoothSumAbsolute` pair, so the gradient stages have a differentiable objective.
- Add a `LossType::log_sum_absolute()` convenience constructor and unit tests (value at `f == t` is 0; monotonicity; smooth → non-smooth convergence as `eps → 0`), following the existing loss tests.

Once available and exposed through the Python binding (`eunoia-py` needs a `loss=` kwarg, tracked separately), eunoia can compete in the logarithmic-L1 group against `venn3` and `matplotlib-set-diagrams("logarithmic")`.

## References

- `crates/eunoia/src/loss.rs` — `LossType` enum and `compute` match arms.
- matplotlib-venn cost: `matplotlib_venn/layout/venn3/cost_based.py` (`WeightedAggregateCost`, default `transform_fn = log(1+x)`).
- matplotlib-set-diagrams objectives: `matplotlib_set_diagrams/_diagram_classes.py` (`cost_function_objective`, the `"logarithmic"` branch).

**Comment by jolars** [comment]
This is included in [`eunoia-capi` v1.1.0](https://github.com/jolars/eunoia/releases/tag/v1.1.0). :tada:

Released by [Versionary](https://github.com/jolars/versionary).
